includes/class-stats.php

```diff
@@ -365,4 +485,15 @@ private function get_stats_for_ajax( $post_id = 0 ) {
 	public function unique_by_post_id( $stat_name, $post_id ) {
 		return $stat_name . '_' . $post_id;
 	}
+
+	/**
+	 * Any page containing a job shortcode is eligible.
+	 *
+	 * @param \WP_Post $post The post.
+	 *
+	 * @return bool
+	 */
+	public function page_has_jobs_shortcode( $post ) {
+		return is_page() && has_shortcode( $post->post_content, 'jobs' );
```

Comment:
Does this work for AJAX calls? If it does and we really want to keep this check, we need to cache it, this sounds very slow to do for each post when logging them in batch. But I don't think we need it or anything in `can_record_stats_for_post`. Even the post type check is a potential slowdown. 

We already have a nonce check for the request, which should prevent most shenanigans. (We can make the nonce action specific to the page, it's just keyed 'ajax-nonce' now)

Reply:
ajax-nonce is required, do you mean adding another one more specific to the stat?

Reply:
The $action of a nonce can be specific to the given action. ([example](https://github.com/Automattic/WP-Job-Manager/blob/7cac449889ee2b2e05e9b1f577209092d02f9763/includes/forms/class-wp-job-manager-form-submit-job.php#L1183))

So if when generating the nonce we are more specific, eg using `'stat_listing_' . $job_id` or `stat_listing_jobs_page`, we don't need to do an extra check that we are on the correct page — the nonce is only accepted if it was generated when rendering the correct page.

Reply:
I removed is page. As for the rest, we can optimize later.